src/cudamatrix/cu-kernels.cu

```diff
@@ -2599,6 +2599,60 @@ static void _diff_softmax(Real* x, const MatrixDim dim, const Real* value,
   }
 }
 
+// Differentiate backward through the log softmax function.
+// "value" is the log softmax output. Does, for each row i,
+// x(i) =  diff(i) - sum(diff(i)) .* exp(value(i))
+// ???(i) is row-vector.
+// CUDA thread layout: one thread block (256 threads) per matrix-row.
+template<typename Real>
+__global__
+static void _diff_log_softmax(Real* x, const MatrixDim dim, const Real* value,
```

Comment:
style comment: I dislike using the name "value" here, even though you document it as the output of log softmax forward propagation. I think "output_value" and "output_value_stride" would be much better names.